tox.ini

```diff
@@ -131,7 +177,68 @@ basepython = python2.7
 
 commands =
     "{toxinidir}/.travis/environment"
-    {toxinidir}/.travis/twistedchecker-diff {posargs:klein}
+
+    "{toxinidir}/.travis/twistedchecker-diff" {posargs:klein}
+
+
+
+##
+# Combine coverage reports
+##
+
+[testenv:coverage_combine]
+
+basepython = python3.5
+
+skip_install = True
+
+deps = coverage
+
+
+whitelist_externals =
+    cp
+
+
+commands =
+    "{toxinidir}/.travis/environment"
+
+    coverage combine --append
+
+    # Copy aside coverage data for each test environment in case we want to look at it later
+    cp "{env:COVERAGE_FILE}" "{envlogdir}/coverage"
+
+    # Run coverage reports, ignore exit status
+    - coverage html --rcfile="{toxinidir}/.coveragerc" -d "{env:COVERAGE_HTML}"
+    - coverage xml --rcfile="{toxinidir}/.coveragerc" -o "{env:COVERAGE_XML}"
+
+    # Don't ignore exit status here; this is our failure status if coverage is insufficient.
+    coverage report --rcfile="{toxinidir}/.coveragerc" --omit "*/test/*"
```

Comment:
I usually prefer to put the --fail-under here, so it's more obvious what conditions we use for failure.